src/model/Publication.ts

```diff
@@ -247,4 +247,133 @@ export default class Publication {
   public positionsByHref(href: string) {
     return this.positions.filter((el: Locator) => el.href === decodeURI(href));
   }
+
+  /**
+   * Fetches the contents to build up the positions manually,
+   * at least for fluid layout pubs
+   */
+  async autoGeneratePositions() {
+    let startPosition = 0;
+    let totalContentLength = 0;
+    const positions = [];
+
+    /**
+     * For each item in the reading order, get its length and calculate
+     * the number of positions in it, then add them all up into the totals.
+     */
+    const promises = this.readingOrder.map(async (link, index) => {
+      // if it is fixed layout, there is no need to fetch, each item is
+      // just a single page.
+      // @aferditamuriqi is this comment correct?
+      if (this.metadata.rendition?.layout === "fixed") {
+        const locator: Locator = {
+          href: link.href,
+          locations: {
+            progression: 0,
+            position: startPosition + 1,
+          },
+          type: link.type,
+        };
+        positions.push(locator);
+        startPosition = startPosition + 1;
+      } else {
+        const href = this.getAbsoluteHref(link.href);
+        // Are we fetching every item in the readingOrder to generate positions?
+        // maybe in the future this can be done in the background?
+        const result = await fetch(href);
+        const length = (await result.blob()).size;
+        link.contentLength = length;
+        totalContentLength += length;
+        const positionLength = 1024;
+        const positionCount = Math.max(1, Math.ceil(length / positionLength));
+        Array.from(Array(positionCount).keys()).forEach((_, position) => {
+          const locator: Locator = {
+            href: link.href,
+            locations: {
+              progression: position / positionCount,
+              position: startPosition + (position + 1),
+            },
+            type: link.type,
+          };
+          positions.push(locator);
+        });
+        startPosition = startPosition + positionCount;
+      }
+
+      // If we are on the last item...
```

Comment:
Instead of testing whether we are on the last item, should this content just be pulled out to run after the `map` loop is finished?